src/plugins/dashboard/public/dashboard_container/embeddable/api/run_save_functions.tsx

```diff
@@ -274,11 +278,28 @@ export async function runInteractiveSave(this: DashboardContainer, interactionMo
         let newTitle = currentState.title;
 
         if (lastSavedId) {
-          const [baseTitle, baseCount] = extractTitleAndCount(newTitle);
-          let copyCount = baseCount + 1;
+          const [baseTitle] = extractTitleAndCount(newTitle);
+
+          const { hits: recentlyCreatedDashboards } = await findDashboards.search({
+            size: 1,
+            search: baseTitle,
+            options: {
+              onlyTitle: true,
+              sort: {
+                sortField: 'created_at',
+                sortOrder: 'desc',
+              },
+            },
+          });
+
+          const [, mostRecentDuplicationId] = extractTitleAndCount(
+            recentlyCreatedDashboards[0].attributes.title
+          );
+
+          let copyCount = mostRecentDuplicationId + 1;
+
           newTitle = `${baseTitle} (${copyCount})`;
```

Comment:
Request the last created dashboard, and use it's duplication Id to create the proposed duplicate title.